### PR TITLE
stdlib: make the library a standalone crate

### PR DESCRIPTION
**Pull request by amtoine**
# Description
as we now have a prelude thanks to #8627, i'd like to work on the structure of the library :yum: 

and i think the first step is to make it a true standalone crate :smirk: 

this PR
- moves all the library from `crates/nu-utils/standard_library/` to `crates/nu-std/`
- moves the `rust` loading code from `src/run.rs` to `crates/nu-std/src/lib.rs`

# User-Facing Changes
```
$nothing
```

# Tests + Formatting
- :green_circle: `toolkit fmt`
- :green_circle: `toolkit clippy`
- :red_circle: `toolkit test` => fails on the unrelated
```
---- commands::each::each_implicit_it_in_block stdout ----
=== stderr

thread 'commands::each::each_implicit_it_in_block' panicked at 'assertion failed: `(left == right)`
  left: `""`,
 right: `"ace"`', crates/nu-command/tests/commands/each.rs:72:5
```
`cargo test --workspace test_stdlib` passes :+1: 
- :green_circle: `toolkit test stdlib`

# After Submitting
```
$nothing
```

### CODE REVIEW

**Comment by fdncred** [comment]
For consistency, I'd call it nu-std. It would be nice to have nu-something since that's the name of all our other crates. We also have to have some rust code I think in order to publish. Not sure what that is though.

**Comment by amtoine** [comment]
and conflict solved :ok_hand:

**Comment by sholderbach** [comment]
Without a `Cargo.toml` (and maybe even either a `lib.rs` or `main.rs`) it is not yet a crate.

There is some potential configuration what non-`.rs` files get bundled when publishing a crate to crates.io https://doc.rust-lang.org/cargo/reference/manifest.html#the-exclude-and-include-fields

Furthermore the code like this is not publishable, as you are referencing assets across crate boundaries right now.

File: src/run.rs [hunk 1]
```diff
@@ -12,7 +12,7 @@ use nu_protocol::{engine::StateWorkingSet, Module, PipelineData, ShellError, Spa
 use nu_utils::utils::perf;
 
 fn get_standard_library() -> &'static str {
-    include_str!("../crates/nu-utils/standard_library/std.nu")
+    include_str!("../crates/nu-std/std.nu")
 }
```
Comment (sholderbach) [inline src/run.rs:16]: this would need to live in the `nu-std` crate and imported here via `nu_std::get_standard_library()`

**Comment by amtoine** [comment]
@sholderbach 

could we call this a "crate" for this very instant?
i'll populate this directory with `rust` source asap, but i thought it would be appropriate to do that into two distinct PRs :yum: 

`crates/nu-std` is not in the `members` of the workspace for now :relieved:

**Comment by amtoine** [comment]
i've changed the title and the message of the PR, they were too vague, sorry for this :cry:

**Comment by sholderbach** [comment]
Alternatively we could place it in some non `crates` asset folder that gets packaged with the `nu` crate. This would limit the standard library to only be used with the `nu` binary as shipped by us but simpler (and more in line with the current test setup "It's just a file).

There are some folks that want to build upon nushell crates. e.g. @stormasm and also the couchbase-shell folks. Moving everything into the `nu` crate would possibly deny them access to `help` and friends. (not the top priority for the mainline nushell project but something to keep in mind)

File: src/run.rs (hunk 1 above)
Comment (amtoine) [reply src/run.rs:16]: yup that's exactly what i have in mind :yum:

**Comment by amtoine** [comment]
i can move all the directory to another place but i'd place it again here in the next PR with the `rust` source... :thinking: 

you tell me :wink:

**Comment by sholderbach** [comment]
My take is, then it should be one PR :P 
The main branch should always be in a state where you only need to update the `Cargo.toml`s to be ready to publish.

**Comment by amtoine** [comment]
> My take is, then it should be one PR :P The main branch should always be in a state where you only need to update the `Cargo.toml`s to be ready to publish.
>
this is a perfect argument, will do that right now :innocent:

**Comment by amtoine** [comment]
@sholderbach @fdncred 

i think we're good now :relieved: 

- i've mirrored the structure of `nu-cli` to write the new `nu-std` library
- i've merged latest `main` to solve the conflicts introduced by #8765 in 8db952f3e01baa50486803fa184c19c697068629

**Comment by amtoine** [comment]
that's fun, we all decided to edit this one piece of `run.rs` at the exact same time :laughing: 

that should be that last conflict merge :relieved:

**Review by sholderbach — APPROVED** [review]
Thanks looks good to me!

**Comment by stormasm** [comment]
@amtoine this looks good to me as well !

Thanks for doing this work...
After this lands and we get a sense for it...

One of the first things I think I would do is try and remove the dependency in this crate from *nu-cli*.
It looks like the only reason you are depending on it is because of a few functions that could probably be moved somewhere else...

The advantage of doing it this way --- is it will enable developers to use our nu-std crate and build their own cli for their application...

Once this lands and a few days have passed of testing etc... 
we can discuss further...

This looks good to me for starters though, excellent effort, I am excited to have this code in its own crate 😄

**Comment by fdncred** [comment]
I think we need the LICENSE file in the crate folder too.

**Comment by stormasm** [comment]
@fdncred thanks for catching this I just added it 😄

**Comment by amtoine** [comment]
@fdncred @stormasm 
very good remarks, and i saw the PRs, so thanks :yum:

**Pull request merged**